client/data/authorizations/actions.ts

```diff
@@ -39,3 +51,59 @@ export function updateAuthorizationsSummary(
 		query,
 	};
 }
+
+export function* submitCaptureAuthorization(
+	id: string,
+	orderId: number,
+	paymentIntentId: string
+): Generator< unknown | Authorization > {
+	try {
+		yield dispatch( STORE_NAME, 'startResolution', 'getAuthorization', [
+			id,
+		] );
+
+		let authorization = yield apiFetch( {
+			path: `/wc/v3/payments/orders/${ orderId }/capture_authorization`,
+			method: 'post',
+			data: {
+				payment_intent_id: paymentIntentId,
+			},
+		} );
+
+		authorization = {
+			authorization_id: id,
+			captured: true,
+		};
+
+		yield updateAuthorization( authorization as Authorization );
+
+		// Need to invalidate the resolution so that the components will render again.
+		yield dispatch(
```

Comment:
I wonder if there's a side effect of invalidating resolutions for selectors on a page where they are not used? Eg. `getAuthorizations` is used on Transactions page but not on Payment Details page. I have not noticed anything, but wanted to confirm.

Reply:
That's a good question, to answer it I dig the[ available docs](https://unfoldingneurons.com/2020/wordpress-data-store-properties-resolvers) for the @wordpres/data package. This is my understanding of how this will work, apologies if the explanation is too verbose.

The `invalidateResolutionForStoreSelector` is a 'special' type of action used by the @wordpres/data to signal the resolution state of a selector/resolver pair. It has its own internal reducer to deal with this type of actions. In this case, the `getAuthorizations` selector will be marked as unresolved in the internal state (aka `metadada`). This means that next time this selector is called, @wordpres/data will know that it needs to resolve it first (fetch from API). Please note that, even if the `getAuthorizations` selector is not used in the Payments Details page, it exists in the store's metadata. In the case it was already resolved (i.e the merchant navigated to `Uncaptured` before) it will be marked as unresolved and refreshed next time is accessed. In the case that it was never resolved, it will be marked as such.

For above reasons, I don't think there will be any undesired side effects. I hope this explanation makes sense.